chips/stm32f446re/src/lib.rs

```diff
@@ -1,6 +1,118 @@
 #![no_std]
 
-pub use stm32f4xx::{self, chip, dbg, dma1, exti, gpio, nvic, rcc, spi, syscfg, tim2, usart};
+pub use stm32f4xx::{chip, dbg, dma1, exti, gpio, nvic, rcc, spi, syscfg, tim2, usart};
 
-pub mod irqs;
 pub mod stm32f446re_nvic;
+
+use cortexm4::generic_isr;
+use stm32f4xx::unhandled_interrupt;
+
+// STM32F446xx has total of 97 interrupts
+// Extracted from `CMSIS/Device/ST/STM32F4xx/Include/stm32f446xx.h`
+// NOTE: There are missing IRQn between 0 and 96
+#[cfg_attr(all(target_arch = "arm", target_os = "none"), link_section = ".irqs")]
+// used Ensures that the symbol is kept until the final binary
```

Comment:
```suggestion
// `used` ensures that the symbol is kept until the final binary. However, as of
// May 2020, due to the compilation process, there must be some other compiled
// code here to make sure the object file is kept around. That means at minimum
// there must be an `init()` function here so that compiler does not just ignore
// the `IRQS` object. See https://github.com/rust-lang/rust/issues/56639 for a
// related discussion.
```

The comment is great, but it'd be nice to have the explanation in both places, so we don't lose track of it in the other board